docs/specifications/tests/Nameserver-TP/nameserver02.md

```diff
@@ -1,61 +1,117 @@
+## Test case identifier
+**NAMESERVER02**
 
-### Objective
+## Objective
 
 EDNS(0) is a mechanism to announce capabilities of a DNS implementation,
 and is now basically required by any new functionality in DNS such as
-DNSSEC.
-Initially standardized in [RFC 2671](https://tools.ietf.org/search/rfc2671)
-and later updated by [RFC 6891](https://tools.ietf.org/search/rfc6891),
-EDNS(0) is a mechanism to announce capabilities of a DNS implementation.
+DNSSEC. EDNS(0) is standardized in [RFC 6891].
+
 This test case checks that all name servers has the capability to do
-EDNS(0).
+EDNS(0) or if not, correctly replies to queries containing EDNS
+(OPT record).
 
-Servers not implementing EDNS(0) returns FORMERR:
+Servers not supporting EDNS(0) must return FORMERR 
+([RFC 6891, section 7]):
 
 > Responders that choose not to implement the protocol extensions
 > defined in this document MUST respond with a return code (RCODE) of
 > FORMERR to messages containing an OPT record in the additional
 > section and MUST NOT include an OPT record in the response.
 
-### Inputs
-
-The domain name to be tested.
-
-### Ordered description of steps to be taken to execute the test case
-
-1. Retrieve all address records for all the name servers using
-   [Method 4](../Methods.md#method-4-obtain-glue-address-records-from-parent) and
-   [Method 5](../Methods.md#method-5-obtain-the-name-server-address-records-from-child),
-   and do recursive lookups for the name servers that are out of bailiwick.
-2. Send a DNS query to each name server IP address querying the SOA record
-   of the domain name with EDNS0 option of payload size ("bufsize")
-   set to 512.
-3. If any answer from step 2 contains a FORMERR RCODE this test case fails.
-4. If the answer does not contain an OPT RR with EDNS version 0, this test
-   case fails.
-
-### Outcome(s)
-
-If any name server returns FORMERR for a query using EDNS(0), or if there is
-no OPT RR with EDNS version 0 in it, this test case fails.
-
-### Special procedural requirements
-
-None.
-
-### Intercase dependencies
-
-None.
-
--------
-
-Copyright (c) 2013, 2014, 2015, IIS (The Internet Infrastructure Foundation)  
-Copyright (c) 2013, 2014, 2015, AFNIC  
-Creative Commons Attribution 4.0 International License
+Servers supporting EDNS(0) must reply with EDNS(0)
+([RFC 6891, section 6.1.1]):
+
+> If an OPT record is present in a received request, compliant
+> responders MUST include an OPT record in their respective responses.
+
+To eliminating the risk of falsely classifying the server as not supporting
+ENDS due e.g. firewall issues, the UDP buffer size is set to 512 bytes 
+(octets).
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Ordered description of steps to be taken to execute the test case
+
+1. Created an SOA query for the *Child Zone* with an OPT record with 
+   EDNS version set to "0" and with EDNS(0) option of payload size ("bufsize")
+   set to 512 and "DO" bit unset.
```

Comment:
@sandoche2k, here I added that the DO bit should be unset. I think that is what is implemented, and maybe not needed, but to make it very clear.

Reply:
OK